storybook/pages/NetworkSelectPopupPage.qml

```diff
@@ -0,0 +1,287 @@
+
+                            spacing: 3
+
+                            Label { text: model.chainName }
+
+                            Row {
+                                spacing: 5
+                                Label { text: `<b>${model.shortName}</b>` }
+                                Label { text: `ID <b>${model.chainId}</b>` }
+                                CheckBox {
+                                    checkState: model.isEnabled ? Qt.Checked : Qt.Unchecked
+                                    tristate: true
+                                    nextCheckState: () => {
+                                        const nextEnabled = (checkState !== Qt.Checked)
+                                        availableNetworks.sourceModel.setProperty(availableNetworks.mapToSource(index), "isEnabled", nextEnabled)
+                                        Qt.callLater(() => { simulatedNimModel.cloneModel(availableNetworks) })
+                                        return nextEnabled ? Qt.Checked : Qt.Unchecked
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    onClicked: allNetworksListView.currentIndex = index
+                }
+            }
+            CheckBox {
+                id: multiSelectionCheckbox
+
+                Layout.margins: 5
+
+                text: "Multi Selection"
+                checked: true
+            }
+            CheckBox {
+                id: testModeCheckbox
+
+                Layout.margins: 5
+
+                text: "Test Networks Mode"
+                checked: false
+                onCheckedChanged: Qt.callLater(simulatedNimModel.cloneModel, availableNetworks)
+            }
+        }
+    }
+
+    SortFilterProxyModel {
+        id: availableNetworks
+
+        // Simulate Nim's way of providing access to data
+        function rowData(index, propName) {
+            return get(index)[propName]
+        }
+
+        sourceModel: NetworksModel.allNetworks
+        filters: ValueFilter { roleName: "isTest"; value: testModeCheckbox.checked; }
+    }
+
+    // Keep a clone so that the UX can be modified without affecting the original model
+    CloneModel {
+        id: simulatedNimModel
+
+        sourceModel: availableNetworks
+
+        roles: ["chainId", "layer", "chainName", "isTest", "isEnabled", "iconUrl", "shortName", "chainColor"]
+        rolesOverride: [{ role: "enabledState", transform: (mD) => {
+                return simulatedNimModel.areAllEnabled(sourceModel)
+                        ? NetworkSelectPopup.UxEnabledState.AllEnabled
+                        : mD.isEnabled
+                            ? NetworkSelectPopup.UxEnabledState.Enabled
+                            : NetworkSelectPopup.UxEnabledState.Disabled
+            }
+        }]
+
+        /// Simulate the Nim model
+        function toggleNetwork(chainId) {
```

Comment:
Could it make sense to have this method in the `NetworksModel.qml` file instead?

Reply:
Well.. maybe the complete `simulatedNimModel`??

Reply:
Great idea. I tried it, but it didn't work because it simulates Nim's network data model, coupled with the NetworkSelectPopup behaviour (see the usage of NetworkSelectPopup.UxEnabledState).

P.S. This is one of the things I wanted to fix with the initial implementation, but I could not succeed in a reasonable time frame due to some Nim data model update issues

Reply:
WDYT if we also add the`enum` in the `NetworksModel.qml` since it is the representation in storybook of our real model, why not having it the same way we have it in nim?

Reply:
It is a great idea, however, I run out of time for this work package but will consider it next time when approaching changes here